### ISSUE TITLE
no-unused-expressions + do{…} expression bug

### ISSUE THREAD

**Issue by phaux** [eslint/eslint#6310]
>The `do { … }` expression executes a block (with one or many statements in it), and the final statement completion value inside the block becomes the completion value of the do expression

```js
console.log(do {
  'hello world'
})
```

This shouldn't cause a warning, but it does:

>Expected an assignment or function call and instead saw an expression. at line 2 col 3

ESLint 2.11.1 using Babel-ESLint parser

**Comment by vitorbal** [comment]
Thanks for the bug report! However, since Do expressions are still a stage 0 proposal, I'm pretty sure the `no-unused-expressions` rule does not take them in consideration.

I think you should open a ticket in the [babel/eslint-plugin-babel](https://github.com/babel/eslint-plugin-babel) repository so they consider adding a "wrapper" rule for `no-unused-expressions` that checks for DoExpressions.

**Comment by Mouvedia** [comment]
For the ones wondering he's talking about http://wiki.ecmascript.org/doku.php?id=strawman:do_expressions
